modules/vmfcore/java/jni/com_intel_vmf_FieldDesc.cpp

```diff
@@ -0,0 +1,237 @@
+#include<string>
+#include<vector>
+#include <memory>
+#include <stdexcept>
+#include "vmf/metadatastream.hpp"
+#include "../com_intel_vmf_FieldDesc.h"
+#include "throwJavaException.hpp"
+
+using namespace vmf;
+
+/*
+ * Class:     com_intel_vmf_FieldDesc
+ * Method:    n_FieldDesc
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_vmf_FieldDesc_n_1FieldDesc__ (JNIEnv *env, jclass)
+{
+    std::shared_ptr<FieldDesc>* p = new std::shared_ptr<FieldDesc>(new FieldDesc ());
+    return (jlong) p;
+}
+
+/*
+ * Class:     com_intel_vmf_FieldDesc
+ * Method:    n_FieldDesc
+ * Signature: (Ljava/lang/String;JZ)J
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_vmf_FieldDesc_n_1FieldDesc__Ljava_lang_String_2IZ(JNIEnv *env, jclass, jstring name, jint type, jboolean isOptional)
+{
+    std::string sName (env->GetStringUTFChars (name, NULL));
+    std::shared_ptr<FieldDesc>* p = new std::shared_ptr<FieldDesc>(new FieldDesc(sName, (Variant::Type) type, (isOptional == JNI_TRUE) ? true : false));
+    return (jlong) p;
+}
+
+/*
+ * Class:     com_intel_vmf_FieldDesc
+ * Method:    n_equals
+ * Signature: (JJ)Z
+ */
+JNIEXPORT jboolean JNICALL Java_com_intel_vmf_FieldDesc_n_1equals (JNIEnv *env, jclass, jlong selfAddr, jlong otherAddr)
+{
+    static const char method_name[] = "FieldDesc::n_1equals";
+    
+    try 
+    {
+        std::shared_ptr<FieldDesc>* self = (std::shared_ptr<FieldDesc>*) selfAddr;
+        std::shared_ptr<FieldDesc>* other = (std::shared_ptr<FieldDesc>*) otherAddr;
+        
+        return ((*(*self) == *(*other)) ? JNI_TRUE : JNI_FALSE);
```

Comment:
actually it's possible to use less parentheses:
`return **self == **other ? JNI_TRUE : JNI_FALSE);`